src/Kestrel.Core/Internal/Infrastructure/KestrelTrace.cs

```diff
@@ -174,7 +174,7 @@ public void RequestBodyMininumDataRateNotSatisfied(string connectionId, string t
             _requestBodyMinimumDataRateNotSatisfied(_logger, connectionId, traceIdentifier, rate, null);
         }
 
-        public void ResponseMininumDataRateNotSatisfied(string connectionId, string traceIdentifier)
+        public virtual void ResponseMininumDataRateNotSatisfied(string connectionId, string traceIdentifier)
```

Comment:
This whole class is now a mix of virtual and non-virtual. I tried to see what would happen if I made everything non-virtual in dev, and the build still passes. It turns out that none of the other methods actually need to be virtual.

Can you make *all* the logging methods either virtual or not for consistency? Thanks.

Reply:
Did you run the whole build or just compilation? My understanding is that they are overridden by Moq mocks. I can make them all virtual for consistency though.

I'm not super happy with that model of verification, but other things used it so I didn't want to rewrite everything.

Reply:
Hang on, this is on KestrelTrace itself. Yeah, I'm going to remove this mock entirely. I want the test logging, hence using KestrelTrace, but the test also verified that the necessary method was called, which I don't think is necessary in this case.